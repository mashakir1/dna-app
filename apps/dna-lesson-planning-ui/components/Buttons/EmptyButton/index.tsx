import styles from "./EmptyButton.module.scss";

export const EmptyButton: ({
  text,
  classNames,
  handleClick,
}: {
  text: string;
  classNames?: string;
  handleClick: () => void;
}) => JSX.Element = ({ text, classNames, handleClick }) => (
  <button
    className={`${styles.emptyButton} ${classNames}`}
    onClick={handleClick}
    type="button"
  >
    {text}
  </button>
);
