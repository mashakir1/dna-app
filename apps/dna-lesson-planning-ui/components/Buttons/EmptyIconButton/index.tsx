import styles from "./EmptyIconButton.module.scss";

export const EmptyIconButton: ({
  icon,
  text,
  classNames,
  handleClick,
}: {
  icon: JSX.Element;
  text: string;
  classNames?: string;
  handleClick: () => void;
}) => JSX.Element = ({ icon, text, classNames, handleClick }) => (
  <button
    className={`${styles.emptyIconButton} ${classNames} flex items-center gap-s`}
    onClick={handleClick}
    type="button"
  >
    {icon}
    <span>{text}</span>
  </button>
);
