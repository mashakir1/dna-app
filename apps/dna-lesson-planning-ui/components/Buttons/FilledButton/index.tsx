// Styles
import styles from "./FilledButton.module.scss";

export const FilledButton: ({
  text,
  handleClick,
  classNames
}: {
  text: string;
  handleClick: () => void;
  classNames?: string;
}) => JSX.Element = ({ text, handleClick, classNames }) => (
  <button
    className={`${styles.filledButton} ${classNames ? classNames : ""}`}
    onClick={handleClick}
    type="button"
  >
    <span>{text}</span>
  </button>
);
