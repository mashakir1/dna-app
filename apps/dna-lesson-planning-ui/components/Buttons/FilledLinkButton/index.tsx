// Imports
import Link from "next/link";

// Styles
import styles from "./FilledLinkButton.module.scss";

export const FilledLinkButton: (props: {
  text: string;
  href: string;
  classNames?: string;
  disabled?: boolean
}) => JSX.Element = ({ text, href, classNames, disabled }) => (
  <Link
    className={`${styles.filledLinkButton} ${classNames ? classNames : ""} ${disabled ? styles.disabled : ""}`}
    href={href}
    aria-disabled={disabled}
  >
    <span>{text}</span>
  </Link>
);
