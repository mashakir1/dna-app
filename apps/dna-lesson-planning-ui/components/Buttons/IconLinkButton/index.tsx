// Imports
import Link from "next/link";

interface IconLinkButtonProps {
  icon: JSX.Element;
  text: string;
  classNames?: string;
  href: string;
}

export const IconLinkButton: (props: IconLinkButtonProps) => JSX.Element = ({
  icon,
  text,
  classNames,
  href,
}) => (
  <Link href={href} className={classNames} aria-label={text}>
    <>
      {icon}
      <span>{text}</span>
    </>
  </Link>
);
