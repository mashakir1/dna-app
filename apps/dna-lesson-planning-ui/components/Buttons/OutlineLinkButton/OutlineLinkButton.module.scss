.outlineLinkButton {
  padding: $spacing-ks $spacing-m;
  border: 1.5px solid $colour_dark_text;
  border-radius: 25px;
  text-align: center;

  &.disabled {
    background-color: $colour_line_grey;
    color: adjust-color($colour_body_copy, $alpha: -0.5);
    border-color: $colour_line_grey;
    pointer-events: none;
  }
}
