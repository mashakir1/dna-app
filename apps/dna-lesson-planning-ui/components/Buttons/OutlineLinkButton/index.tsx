// Imports
import Link from "next/link";

// Styles
import styles from "./OutlineLinkButton.module.scss";

interface OutlineLinkButtonProps {
  text: string;
  classNames?: string;
  href: string;
  disabled?: boolean;
}

export const OutlineLinkButton: (props: OutlineLinkButtonProps) => JSX.Element = ({
  text, classNames, href, disabled
}) => {
  return (
    <Link href={href} className={`${styles.outlineLinkButton} ${classNames} ${disabled ? styles.disabled : ""}`}>
      <span>{text}</span>
    </Link>
  );
};
