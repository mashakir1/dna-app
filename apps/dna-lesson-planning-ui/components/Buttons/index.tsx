export { IconLinkButton } from "./IconLinkButton";
export { OutlineLinkButton } from "./OutlineLinkButton";
export { FilledButton } from "./FilledButton";
export { FilledLinkButton } from "./FilledLinkButton";
export { EmptyButton } from "./EmptyButton";
export { EmptyIconButton } from "./EmptyIconButton";
