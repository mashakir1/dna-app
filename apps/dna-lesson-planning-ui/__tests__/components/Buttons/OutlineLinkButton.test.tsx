import { render } from "@testing-library/react";
import "@testing-library/jest-dom";

import { OutlineLinkButton } from "components/Buttons/OutlineLinkButton";

describe("OutlineLinkButton", () => {
  const text = "OutlineLinkButton";
  const href = "/plan/subject";
  it("should render button with back arrow", () => {
    const { container } = render(<OutlineLinkButton text={text} href={href} />);
    expect(container).toMatchSnapshot();
  });
  it("should render button with book icon", () => {
    const { container } = render(<OutlineLinkButton text={text} href={href} />);
    expect(container).toMatchSnapshot();
  });
});
