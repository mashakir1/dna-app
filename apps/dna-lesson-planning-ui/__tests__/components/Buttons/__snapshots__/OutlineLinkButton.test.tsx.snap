// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OutlineLinkButton should render button with back arrow 1`] = `
<div>
  <span>
    OutlineLinkButton
  </span>
</div>
`;

exports[`OutlineLinkButton should render button with book icon 1`] = `
<div>
  <span>
    OutlineLinkButton
  </span>
</div>
`;
